src/main/java/com/bmw/hmm_lib/ViterbiAlgorithm.java

```diff
@@ -157,7 +159,9 @@ private boolean hmmBreak(Map<S, Double> message) {
     private ForwardStepResult forwardStep(HmmProbabilities<S, O> hmmProbabilities,
             TimeStep<S, O> prevTimeStep, TimeStep<S, O> curTimeStep, Map<S, Double> message) {
         final ForwardStepResult result = new ForwardStepResult(curTimeStep.candidates.size());
-        assert( !prevTimeStep.candidates.isEmpty());
```

Comment:
The problem with assert is that it is ignored in production mode (i.e. without `-ea`) making it more or less meaningless. So I would propose to completely remove it or keep the 'if'.